pmd-core/src/main/java/net/sourceforge/pmd/internal/Slf4jSimpleConfiguration.java

```diff
@@ -43,22 +40,39 @@ public static void reconfigureDefaultLogLevel(Level level) {
             Method initMethod = simpleLoggerClass.getDeclaredMethod("init");
             initMethod.setAccessible(true);
             initMethod.invoke(null);
+
+            // Call SimpleLoggerFactory.reset() by reflection.
+            Method resetMethod = loggerFactoryClass.getDeclaredMethod("reset");
+            resetMethod.setAccessible(true);
+            resetMethod.invoke(loggerFactory);
```

Comment:
This call is needed with slf4j 1.7.x - here slf4j uses StaticBinding and only one instance of SimpleLoggerFactory is ever created. With slf4j 2.x a new instance will be created, when the logging is reinitialized (which is triggered by (Pmd)LoggerFactoryFriend.reset()).